RIOT-OS/RIOT
# test_utils: add UDP benchmark

### Contribution description

I needed a tool to generate configurable load on the network so I cobbled this together.
Maybe it's useful for others too.


### Testing procedure

Add the `benchmark_udp` module to your application

    USEMODULE += benchmark_udp

e.g.:

    examples/gnrc_networking$ USEMODULE='benchmark_udp' make

Start the benchmark process by specifying the IP of the host running the benchmark server

    benchmark start fd00:dead:beef::1

Build and start the benchmark server on your local machine

    make -C dist/tools/benchmark_udp
    dist/tools/benchmark_udp/bin/benchmark_server :: 12345

Observe the results:

```
host                       	bandwith	num RX	num TX	rate	RTT
2001:db9::d44c:5fff:fe16:1ebc	 439 b/s	111	111	100%	480 µs
2001:db9::c44:6ff:fe3b:50cf  	 474 b/s	186	186	100%	617 µs
```

You can use the `-i <µs>` option to adjust the interval in which clients will send packets.
With the `-s <bytes>` option you can adjust the size of the packets. E.g.

    dist/tools/benchmark_udp/bin/benchmark_server :: 12345 -i 100 -s 100

to send packets with 100 byte payload every 100 µs

```
host                         	bandwith	num RX	num TX	rate	RTT
2001:db9::d44c:5fff:fe16:1ebc	318671 b/s	48449	55857	86%	301 µs
2001:db9::c44:6ff:fe3b:50cf  	263959 b/s	40247	42520	94%	404 µs
```

### Issues/PRs references
